src/sentry/dynamic_sampling/rules/base.py

```diff
@@ -47,9 +52,9 @@ def _get_rules_of_enabled_biases(
     return rules
 
 
-def generate_rules(project: Project) -> List[PolymorphicRule]:
+def generate_rules(project: Project) -> Sequence[PolymorphicActiveRule]:
```

Comment:
Why would you change the return to a `Sequence`? Is there a specific reasoning?

Reply:
mypy started complaining because `List` is invariant while `Sequence` is covariant ([docs](https://mypy.readthedocs.io/en/stable/generics.html#:~:text=List%20is%20an%20invariant%20generic%20type.%20Naively%2C%20one%20would%20think%20that%20it%20is%20covariant%2C%20like%20Sequence)). With your suggestion to just remove `active`, I was able to revert this change though.